odoo/addons/base/models/ir_actions.py

```diff
@@ -564,10 +564,18 @@ def run(self):
                  return action
         """
         res = False
-        for action in self:
+        for action in self.sudo():
             action_groups = action.groups_id
-            if action_groups and not (action_groups & self.env.user.groups_id):
-                raise AccessError(_("You don't have enough access rights to run this action."))
+            if action_groups:
+                if not (action_groups & self.env.user.groups_id):
+                    raise AccessError(_("You don't have enough access rights to run this action."))
+            else:
+                try:
+                    self.env[action.model_name].check_access_rights("write")
+                except AccessError:
+                    _logger.warning("Forbidden server action %r executed while the user %s does not have access to %s." % (
+                        action.name, self.env.user.login, action.model_name,
+                    ))
```

Comment:
```suggestion
            elif not self.env[action.model_name].check_access_rights("write", raise_exception=False):
                  _logger.warning("Forbidden server action %r executed while the user %s does not have access to %s." % (
                      action.name, self.env.user.login, action.model_name,
                  ))
```
Also do we really want to just warn and execute the action anyway?

And logging calls should leave the formatting to logging.

Reply:
> Also do we really want to just warn and execute the action anyway?

No this is why there is a `raise` below

Reply:
note: you added a raise in a different commits so that's fine.

Reply:
> > Also do we really want to just warn and execute the action anyway?
> 
> No this is why there is a `raise` below

I commented by reading the commits individually so I didn't see the raise until a bit later.

Reply:
Ok I need to squash these commits :+1: